src/tls/openssl.c

```diff
@@ -528,11 +529,25 @@ static int tls_net_handshake(struct flb_tls *tls,
             return -1;
         }
 
+        target_host = vhost;
         if (vhost != NULL) {
```

Comment:
Could you please add an integer field to `tls_context` that acts as a boolean and is initialized in [tls_context_create](https://github.com/fluent/fluent-bit/blob/f16d1c02fe92fb27b047a3da01e2f4b1d13d8ba6/src/tls/openssl.c#L260) with the value of `verify`?

That way we could combine that and the fact that `vhost` is not `NULL` to determine if we want to enforce certificate pinning.